addons/mail/static/src/js/services/chat_manager.js

```diff
@@ -1115,19 +1114,97 @@ var ChatManager =  AbstractService.extend({
-                    }
+                self.cannedResponses.push(_.pick(s, ['id', 'source', 'substitution']));
+            });
+            // emojis
+
+            var lastadded = null;
+            self.emojiUnicodes={
+                ":)":"😊", ":-)":"😊","=)":"😊", ":]":"😊",
+                ":D":"😃",":-D":"😃","=D":"😃",
+                "xD":"😆","XD":"😆",
+                "x'D":"😂",
+                ";)":"😉",";-)":"😉",
+                "B)":"😎","8)":"😎","B-)":"😎","8-)":"😎",
+                ";p":"😜",";P":"😜",
+                ":p":"😋",":P":"😋",":-p":"😋",":-P":"😋","=P":"😋",
+                "xp":"😝","xP":"😝",
+                "o_o":"😳",
+                ":|":"😐",":-|":"😐",
+                ":/":"😕",":-/":"😕",
+                ":(":"😞",
+                ":@":"😱",
+                ":O":"😲",":-O":"😲",":o":"😲",":-o":"😲",
+                ":'o":"😨",  
+                "3:(":"😠",">:(":"😠","3:":"😠",
+                "3:)":"😈",">:)":"😈",
+                ":*":"😘",":-*":"😘",
+                "o:)":"😇",
+                ":'(":"😢",
+                ":'-(":"😭",":\"(":"😭",
+                "&lt;3":"❤️",":heart":"❤️",
+                ":heart_eyes":"😍",
+                ":turban":"👳",
+                ":+1":"👍",
+                ":-1":"👎",
+                ":ok":"👌",
+                ":poop":"💩",
+                ":no_see":"🙈",
+                ":no_hear":"🙉",
+                ":no_speak":"🙊",
+                ":bug":"🐞",
+                ":kitten":"😺",
+                ":bear":"🐻",
+                ":snail":"🐌",
+                ":boar":"🐗",
+                ":clover":"🍀",
+                ":sunflower":"🌹",
+                ":fire":"🔥",
+                ":sun":"☀️",
+                ":partly_sunny:":"⛅️",
+                ":rainbow":"🌈",
+                ":cloud":"☁️",
+                ":zap":"⚡️",
+                ":star":"⭐️",
+                ":cookie":"🍪",
+                ":pizza":"🍕",  
+                ":hamburger":"🍔", 
+                ":fries":"🍟",
+                ":cake":"🎂",
+                ":cake_part":"🍰",
+                ":coffee":"☕️",
+                ":banana":"🍌",
+                ":sushi":"🍣",
+                ":rice_ball":"🍙",
+                ":beer":"🍺",
+                ":wine":"🍷",
+                ":cocktail":"🍸",
+                ":tropical":"🍹",
+                ":beers":"🍻",
+                ":ghost":"👻",
+                ":skull":"💀",
+                ":et":"👽",":alien":"👽",
+                ":party":"🎉",
+                ":trophy":"🏆",
+                ":key":"🔑",
+                ":pin":"📌",
+                ":postal_horn":"📯",
+                ":music":"🎵",
+                ":trumpet":"🎺",
+                ":guitar":"🎸",
+                ":soccer":"⚽️",
+                ":football":"🏈",
+                ":8ball":"🎱",
+                ":clapper":"🎬",
+                ":microphone":"🎤"
+                };
+            _.each(_.keys(self.emojiUnicodes), function (key) {
```

Comment:
`_.each(self.emojiUnicodes, function (unicode, key) {`

_.each can be used on objects